x509/test/Test/Cardano/X509/Configuration.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE TemplateHaskell #-}
+module Test.Cardano.X509.Configuration (tests) where
+
+import           Cardano.X509.Configuration
+import qualified Data.List.NonEmpty
+import           Hedgehog (Gen, Property)
+import qualified Hedgehog as H
+import qualified Hedgehog.Gen
+import           Hedgehog.Range
+import           Test.Pos.Util.Tripping
+import           Universum
+
+roundTripTLSConfiguration :: Property
+roundTripTLSConfiguration = roundTripsAesonYamlShow 100 genTlsConfig
+
+genTlsConfig :: Gen TLSConfiguration
+genTlsConfig = TLSConfiguration <$> genCertConfig <*> genServerConfig <*> genClients
+
+genCertConfig :: Gen CertConfiguration
+genCertConfig = CertConfiguration <$> genString <*> genString <*> genInt
+
+genServerConfig :: Gen ServerConfiguration
+genServerConfig = ServerConfiguration <$> genCertConfig <*> genNonEmptyString
+
+genClients :: Gen [CertConfiguration]
+genClients = Hedgehog.Gen.list (linearFrom 0 0 10) genCertConfig
+
+genString :: Gen String
+genString = Hedgehog.Gen.string (linearFrom 0 0 10) Hedgehog.Gen.ascii
+
+genNonEmptyString :: Gen (NonEmpty String)
+genNonEmptyString = pure $ Data.List.NonEmpty.fromList [ "foo.com" ] -- TODO
```

Comment:
Instead of  "foo.com", is it possible to use `genString` instead? Also what is the `TODO` comment referring too? Could you be more explicit here?

Reply:
not sure how to combine the monadic generators with `NonEmpty` to make a list of variable length and random contents